File: ruby/lib/fizz_buzz.rb
class FizzBuzz
  # attr_reader :numbers

  def value_of number
  	return (number % 15 == 0) ? "FizzBuzz" : (number % 3 == 0 ? "Fizz" : (number % 5 == 0 ? "Buzz" : number.to_s))
  end
end

File: java/FizzBuzz/src/FizzBuzz.java
import java.util.ArrayList;

public class FizzBuzz {
    private ArrayList<String> numbers;

    public FizzBuzz(int count) {
        numbers = new ArrayList<String>();
        for (int number = 0; number <= count; number++) {
            numbers.add(valueOf(number));
        }
    }

    public ArrayList<String> numbers() {
        return numbers;
    }

    public String valueOf(int number) {
        String value = "";
        if (number % 3 == 0) {
            value +="Fizz";
        }
        if (number % 5 == 0) {
            value += "Buzz";
        }
        return value == "" ? String.valueOf(number) : value;
    }
}
